sbi/src/api/tee_guest.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::TeeGuestFunction::*;
+use crate::{ecall_send, Result, SbiMessage};
+
+/// Registers an emulated MMIO region. Future accesses in the specified address range will trap
+/// to the host, allowing it to emulated the access.
+pub fn add_emulated_mmio_region(addr: u64, len: u64) -> Result<()> {
+    let msg = SbiMessage::TeeGuest(AddEmulatedMmioRegion { addr, len });
+    // Safety: AddEmulatedMmioRegion does not access our memory.
```

Comment:
Is this safe because passing any address that is already accessible to the guest will be failed with an error? Worth mentioning that as it isn't obvious what the behavior is if passed an address to an existing object.

Reply:
Yes. Updated the "Safety" and function-level comment.